app/controllers/coronavirus_form/business_details_controller.rb

```diff
@@ -1,39 +1,32 @@
 # frozen_string_literal: true
 
 class CoronavirusForm::BusinessDetailsController < ApplicationController
-  before_action :check_first_question_answered, only: :show
+  before_action :check_first_question_answered, except: :submit
 
   REQUIRED_FIELDS = %w(company_name).freeze
   TEXT_FIELDS = %w(company_name company_number).freeze
 
-  def show
-    render "coronavirus_form/#{PAGE}"
-  end
-
   def submit
     @business_details = sanitized_business_details(params)
     session[:business_details] = @business_details
 
-    invalid_fields = validate_field_response_length(PAGE, TEXT_FIELDS) +
+    invalid_fields = validate_field_response_length(controller_name, TEXT_FIELDS) +
       validate_fields(@business_details)
 
     if invalid_fields.any?
       flash.now[:validation] = invalid_fields
-      render "coronavirus_form/#{PAGE}"
+      render controller_path
     else
-      redirect_to controller: session["check_answers_seen"] ? "coronavirus_form/check_answers" : "coronavirus_form/#{NEXT_PAGE}", action: "show"
+      redirect_to session["check_answers_seen"] ? check_your_answers_path : contact_details_path
```

Comment:
Oh dear! I know this isn't part of this PR, but inconsistent use of a ternary operation. The other controllers use an `elsif`.